src/System.Numerics.Tensors/src/System/Numerics/Tensors/ITensor.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Tensors
+{
+    public interface ITensor<T>
+    {
+        ReadOnlySpan<int> Dimensions { get; }
+        bool IsReversedStride { get; }
+        long Length { get; }
+        int Rank { get; }
+        ReadOnlySpan<int> Strides { get; }
```

Comment:
I think `Strides` is probably an `IDenseTensor` concept; as you probably won't be working with strides otherwise.

I'm also not convinced we need to keep track of the `Strides` ourselves...

Reply:
Strides is important to map indices to buffer offsets.  It could be a DenseTensor concept only, though I think I was using it as a generalization of mapping indices to linear offsets in the other implementations.

Reply:
> Strides is important to map indices to buffer offsets

I agree that it is an important concept. However, I would think that the common case is that you can cheaply compute the stride as part of computing the linear offset.

That is, you would generally have an indice per rank (unless you are slicing and returning a sub-tensor), in which case you already have to compute the linear offset by multiplying the various indices by their respective strides. Computing the stride dynamically, at that point, isn't much more expensive (and looks to be what other common implementations are doing).

Reply:
I agree you can compute the strides using IsReversedStride and Dimensions as we do [here](https://github.com/dotnet/corefx/blob/master/src/System.Numerics.Tensors/src/System/Numerics/Tensors/ArrayUtilities.cs#L71), but we would need to do that so often and it never changes for the lifetime of the Tensor that I felt like it would make sense to keep it as I felt the CPU vs Memory tradeoff was in favor of the memory here.  The idea behind exposing it is that it would be needed for folks to write efficient math algos that dealt with the raw buffer and avoided method calls.  I originally exposed this [here](https://github.com/dotnet/corefxlab/commit/0e5bbc3e4ad38185fa7e6e2f556b6e17499707ba#diff-7596a0b9e908acb2e6dee67e701749a4R368) but never did use the publicly exposed property.

I think one of the ideas I had at the time was that a sliced tensor could use the same backing memory as a normal densetensor with merely different strides/dimensions (where strides would actually skip blocks of memory and thus could not be purely calculated from dimensions).  I don't think we actually went with that design and instead a slice was exposed as [its own tensor type](https://github.com/dotnet/corefxlab/pull/1927/files#diff-c16b49ee488bcc01e5bf7d9ad45295f2R10) that didn't directly expose its backing memory.

Reply:
> I felt like it would make sense to keep it as I felt the CPU vs Memory tradeoff was in favor of the memory here

At the same time, I wasn't able to find other implementations that are exposing this, so I'm not sure its worthwhile to be part of the "contract".